// @azure/storage-file-share/dist-esm/storage-file-share/src/index.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { RestError } from "@azure/core-rest-pipeline";
export * from "./AccountSASPermissions";
export * from "./AccountSASResourceTypes";
export * from "./AccountSASServices";
export { generateAccountSASQueryParameters, } from "./AccountSASSignatureValues";
export * from "./FileSASPermissions";
export { generateFileSASQueryParameters } from "./FileSASSignatureValues";
export * from "./Clients";
export * from "./ShareSASPermissions";
export * from "../../storage-blob/src/credentials/AnonymousCredential";
export * from "../../storage-blob/src/credentials/Credential";
export * from "../../storage-blob/src/credentials/StorageSharedKeyCredential";
export { StorageFileAudience, getFileServiceAccountAudience, } from "./models";
export * from "./FileSystemAttributes";
export { BaseRequestPolicy } from "../../storage-blob/src/policies/RequestPolicy";
export { Pipeline, isPipelineLike, newPipeline, StorageOAuthScopes, } from "./Pipeline";
export * from "../../storage-blob/src/policies/AnonymousCredentialPolicy";
export * from "../../storage-blob/src/policies/CredentialPolicy";
export * from "./StorageRetryPolicyFactory";
export * from "../../storage-blob/src/policies/StorageSharedKeyCredentialPolicy";
export * from "../../storage-blob/src/StorageBrowserPolicyFactory";
export * from "./ShareServiceClient";
export * from "./SASQueryParameters";
export * from "./generatedModels";
export { RestError };
export { logger } from "./log";
